src/utils.jl

```diff
@@ -33,6 +33,7 @@ end
 map_params_as_tuple_fallback(f, ::Type{T}) where {T<:Tup} = map(f, fieldtypes(T))
 
 buildfromschema(initializer::F, ::Type{T}) where {F, T} = buildfromschema(initializer, T, staticschema(T))
+buildfromschema(initializer, ::Type, ::Type{NamedTuple{(), Tuple{}}}) = error("Only structs with fields are supported")
```

Comment:
Maybe best to add the error in the [inner constructor](https://github.com/JuliaArrays/StructArrays.jl/blob/3fe67c7b28b9a17dfd7aef694cc94a478451cfff/src/structarray.jl#L17)? It has an explicit check for the length of columns, one could just throw if there are none.

Also, I think one should probably remove the special overloads for `size` and `axes` [here](https://github.com/JuliaArrays/StructArrays.jl/blob/3fe67c7b28b9a17dfd7aef694cc94a478451cfff/src/structarray.jl#L336).

Reply:
They are only called to allow the weird `StructArray` with no field to exist and be a vector of length `0`. If  this no longer exists, they won't be called at all and should be removed IMO.